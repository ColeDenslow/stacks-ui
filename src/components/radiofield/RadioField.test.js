import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RadioField from './RadioField';

describe('RadioField', () => {
  const onChange = () => 'onChange';
  const onClick = () => 'onClick';

  test('RadioField renders with basic props', () => {
    const output = mount(<RadioField
      className='the-class'
      data-additional-prop='additional-prop'
      onChange={onChange}
      onClick={onClick}
      tabIndex='1'
      name='name'
      value='value'
      disabled
      checked
      messageType='alert'
      messageText='message'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('RadioField render disabled when disabled is true', () => {
    const output = shallow(<RadioField disabled />);
    expect(toJson(output)).toMatchSnapshot();
  });
});
