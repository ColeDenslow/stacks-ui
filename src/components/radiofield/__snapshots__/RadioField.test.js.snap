// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioField RadioField render disabled when disabled is true 1`] = `
<Field
  className="stx-field--with-radio-field"
  disabled={true}
>
  <div
    className="stx-radio-field"
  >
    <Radio
      className="stx-radio-field__radio"
      disabled={true}
      id="stxField2"
    />
    <label
      className="stx-radio-field__label"
      htmlFor="stxField2"
    />
  </div>
</Field>
`;

exports[`RadioField RadioField renders with basic props 1`] = `
<RadioField
  checked={true}
  className="the-class"
  data-additional-prop="additional-prop"
  disabled={true}
  messageText="message"
  messageType="alert"
  name="name"
  onChange={[Function]}
  onClick={[Function]}
  tabIndex="1"
  value="value"
>
  <Field
    className="stx-field--with-radio-field the-class"
    disabled={true}
    messageText="message"
    messageType="alert"
  >
    <div
      className="stx-field stx-field--alert stx-field--is-disabled stx-field--with-radio-field the-class"
    >
      <div
        className="stx-radio-field"
      >
        <Radio
          checked={true}
          className="stx-radio-field__radio"
          data-additional-prop="additional-prop"
          disabled={true}
          id="stxField1"
          name="name"
          onChange={[Function]}
          onClick={[Function]}
          tabIndex="1"
          value="value"
        >
          <span
            className="stx-radio stx-radio--is-disabled stx-radio--is-checked stx-radio-field__radio"
          >
            <input
              aria-checked={true}
              aria-disabled={true}
              checked={true}
              className="stx-radio__input"
              data-additional-prop="additional-prop"
              disabled={true}
              id="stxField1"
              name="name"
              onChange={[Function]}
              onClick={[Function]}
              role="radio"
              tabIndex="1"
              type="radio"
              value="value"
            />
            <span
              className="stx-radio__box"
            />
          </span>
        </Radio>
        <label
          className="stx-radio-field__label"
          htmlFor="stxField1"
        />
      </div>
      <div
        className="stx-field__message"
      >
        message
      </div>
    </div>
  </Field>
</RadioField>
`;
